core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpReceiverTest.java

```diff
@@ -1258,6 +1261,144 @@ public void testWriterInsertNewSymbolsIntoTableWithExistingSymbols() throws Exce
+        Assume.assumeTrue(walEnabled);
+        configOverrideMaxUncommittedRows(2);
+        configOverrideWalSegmentRolloverRowCount(2);
+        FilesFacade filesFacade = new TestFilesFacadeImpl() {
+            @Override
+            public int openRW(LPSZ name, long opts) {
+                if (Chars.endsWith(name, "weather~2" + Files.SEPARATOR + "wal1" + Files.SEPARATOR + "1.lock")) {
+                    mayDrainWalQueue();
+                    renameTable("weather", "meteorology");
+                }
+                return super.openRW(name, opts);
+            }
+        };
+
+        runInContext(filesFacade, (receiver) -> {
+            String lineData = "weather,location=west1 temperature=10 1465839830100400200\n" +
+                    "weather,location=west2 temperature=20 1465839830100500200\n" +
+                    "weather,location=east3 temperature=30 1465839830100600200\n" +
+                    "weather,location=west4,source=sensor1 temp=40 1465839830100700200\n" + // <- this is where the split should happen
+                    "weather,location=east5,source=sensor2 temp=50 1465839830100800200\n" +
+                    "weather,location=west6,source=sensor3 temp=60 1465839830100900200\n";
+            send(receiver, lineData, "weather");
+
+            lineData = "weather,location=north,source=sensor4 temp=70 1465839830101000200\n";
+            send(receiver, lineData, "weather");
+            lineData = "meteorology,location=south temperature=80 1465839830101000200\n";
+            send(receiver, lineData, "meteorology");
+
+            mayDrainWalQueue();
+            String expected = "location\ttemperature\ttimestamp\tsource\ttemp\n" +
+                    "west1\t10.0\t2016-06-13T17:43:50.100400Z\t\tNaN\n" +
+                    "west2\t20.0\t2016-06-13T17:43:50.100500Z\t\tNaN\n" +
+                    "east3\t30.0\t2016-06-13T17:43:50.100600Z\t\tNaN\n" +
+                    "west4\tNaN\t2016-06-13T17:43:50.100700Z\tsensor1\t40.0\n" +
+                    "south\t80.0\t2016-06-13T17:43:50.101000Z\t\tNaN\n";
+            assertTable(expected, "meteorology");
+
+            expected = "location\tsource\ttemp\ttimestamp\n" +
+                    "east5\tsensor2\t50.0\t2016-06-13T17:43:50.100800Z\n" +
+                    "west6\tsensor3\t60.0\t2016-06-13T17:43:50.100900Z\n" +
+                    "north\tsensor4\t70.0\t2016-06-13T17:43:50.101000Z\n";
+            assertTable(expected, "weather");
+
+        }, false, 250);
+    }
+
+    @Test
+    public void testRenameTableSameMeta() throws Exception {
+        Assume.assumeTrue(walEnabled);
+        configOverrideMaxUncommittedRows(2);
+        configOverrideWalSegmentRolloverRowCount(2);
+        FilesFacade filesFacade = new TestFilesFacadeImpl() {
+            @Override
+            public int openRW(LPSZ name, long opts) {
+                if (Chars.endsWith(name, "weather~2" + Files.SEPARATOR + "wal1" + Files.SEPARATOR + "1.lock")) {
```

Comment:
Can this `~2` be not hardcoded? TableId or directory format can be changed in the future and this test become unstable.

Reply:
Fixed in https://github.com/questdb/questdb/pull/2926/commits/0f1dde3cf393ebfbf8b6d2d8a2a7a3b81d51dfd6